tests/python/pants_test/integration/pep_561_integration_test.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from glob import glob
+from textwrap import dedent
+
+from pants.base.build_environment import get_buildroot
+from pants.testutil.pants_integration_test import PantsResult, run_pants, setup_tmpdir
+
+
+def typecheck_file(filename: str) -> PantsResult:
+    return run_pants(
+        [
+            "--backend-packages=["
+            "'pants.backend.python',"
+            "'pants.backend.python.typecheck.mypy'"
+            "]",
+            "typecheck",
+            filename,
+        ]
+    )
+
+
+def test_typecheck_works() -> None:
+    project = {
+        "proj1/BUILD": "python_library()",
+        "proj1/ok.py": dedent(
+            """\
+            def dummy() -> int:
+                return 42
+            """
+        ),
+        "proj1/err.py": dedent(
+            """\
+            def dummy() -> int:
+                return "not an int"
+            """
+        ),
+    }
+
+    with setup_tmpdir(project) as tmpdir:
+        ok_result = typecheck_file(f"{tmpdir}/proj1/ok.py")
+        err_result = typecheck_file(f"{tmpdir}/proj1/err.py")
+
+    ok_result.assert_success()
+    err_result.assert_failure()
+
+
+def test_type_stubs() -> None:
+    wheel = glob(f"{get_buildroot()}/pantsbuild.pants.testutil-*.whl")[0]
+    project = {
+        "proj1/BUILD": dedent(
+            f"""\
+            python_requirement_library(
+                name="pants",
+                requirements=["Pants @ file://{wheel}"],
+            )
+            python_library()
+            """
+        ),
+        "proj1/ok.py": dedent(
+            """\
+            # import should be ok
+            from pants.testutil.option_util import create_subsystem
+            """
+        ),
+        "proj1/err.py": dedent(
+            """\
+            from pants.testutil.option_util import create_subsystem
+
+            def dummy() -> None:
+                # wrong args should not be ok
+                create_subsystem()
+            """
+        ),
+    }
+
+    with setup_tmpdir(project) as tmpdir:
+        ok_result = typecheck_file(f"{tmpdir}/proj1/ok.py")
```

Comment:
As far as I can tell,  this is failing because the wheel you're grabbing is for Python 3.{7,8,9} but the `typecheck_file` helper does not run Pants with interpreter constraints constraining it as such and it's picking a Python 3.6. Pay careful attention to the `--dest` option value:
```
E         	stderr:
E         	ERROR: pantsbuild.pants.testutil-2.8.0.dev0-py37.py38.py39-none-any.whl is not a supported wheel on this platform.
E         	pid 35909 -> /home/runner/.cache/pants/named_caches/pex_root/venvs/813aff41909de48f9fc4117c2598a7e12f48f668/44bf65ec4b573a3d710ed08dd3dbbf662b72808e/pex --disable-pip-version-check --no-python-version-warning --exists-action a --isolated -q --cache-dir /home/runner/.cache/pants/named_caches/pex_root --log /tmp/process-executionDIyox8/.tmp/tmpgqy7v_qg/pip.log download --dest /tmp/process-executionDIyox8/.tmp/tmpylkk2n66/opt.hostedtoolcache.Python.3.6.14.x64.bin.python3.6 Pants@ file:///tmp/process-executionsHKkKw/pantsbuild.pants.testutil-2.8.0.dev0-py37.py38.py39-none-any.whl --index-url https://pypi.org/simple/ --retries 5 --timeout 15 exited with 1 and STDERR:
```

Reply:
Oh... thanks for pointing that out.